geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -89,8 +92,31 @@ object AccumuloDataStoreFactory {
     val instance = instanceIdParam.lookup[String](params)
     val user = userParam.lookup[String](params)
     val password = passwordParam.lookup[String](params)
+    val keytabPath = keytabPathParam.lookup[String](params)
+
+    var usingKerberos = false
+
+    // Build authenticaton token according to how we are authenticating
+    val authToken : AuthenticationToken = if (password != null ) {
+      new PasswordToken(password.getBytes("UTF-8"))
+    } else if (keytabPath != null && !useMock) {
+      usingKerberos = true
+      // Get ctor for principal, keytab path and replace current user
+      val ctor = Class.forName("org.apache.accumulo.core.client.security.tokens.KerberosToken")
```

Comment:
I think we've dropped official support for 1.6, although we'd like to keep unofficial support. In general if it compiles with 1.7, but will still work with a 1.6 runtime we should be ok. I think that's the case here, since this block of code won't get hit in 1.6. Alternatively, we'd tried to keep all our accumulo versioned code in the `AccumuloVersion` class, which does use reflection - maybe you could move this there.